packages/allow-scripts/src/index.js

```diff
@@ -1,70 +1,227 @@
-    await runAllScriptsForEvent({ event: 'postinstall', packages: allowedPackagesWithLifecycleScripts })
+    await runAllScriptsForEvent({ event: 'postinstall', packages: allowedPackagesWithScriptsLifecycleScripts })
   } else {
-    console.log('no allowed scripts found in configuration')
+    console.log('no allowed lifecycle scripts found in configuration')
   }
 
   // run scripts in top-level package
   console.log('running lifecycle scripts for top level package')
   await runScript({ event: 'install', path: rootDir })
   await runScript({ event: 'postinstall', path: rootDir })
   await runScript({ event: 'prepublish', path: rootDir })
+  // TODO: figure out if we should be doing this:
   await runScript({ event: 'prepare', path: rootDir })
 }
 
-async function runAllScriptsForEvent ({ event, packages }) {
+async function setDefaultConfiguration({ rootDir }) {
+  const conf = await loadAllPackageConfigurations({ rootDir })
+  const {
+    configs: {
+      lifecycle,
+      bin
+    },
+    somePoliciesAreMissing
+  } = conf
+
+  console.log('\n@lavamoat/allow-scripts automatically updating configuration')
+
+  if (!somePoliciesAreMissing) {
+    console.log('\nconfiguration looks good as is, no changes necesary')
+    return
+  }
+
+  console.log('\nadding configuration for missing packages:')
+
+  lifecycle.missingPolicies.forEach(pattern => {
+    console.log(`- lifecycle ${pattern}`)
+    lifecycle.allowConfig[pattern] = false
+  })
+
+  if(bin.somePoliciesAreMissing) {
+    bin.allowConfig = prepareBinScriptsPolicy(bin.binCandidates)
+  }
+
+  // update package json
+  await savePackageConfigurations({
+    rootDir,
+    conf
+  })
+}
+
+async function printPackagesList({ rootDir }) {
+  const {
+    configs: {
+      bin,
+      lifecycle
+    }
+  } = await loadAllPackageConfigurations({ rootDir })
+
+  printPackagesByBins(bin)
+  printPackagesByScriptConfiguration(lifecycle)
+}
+
+
+function printMissingPoliciesIfAny({ missingPolicies = [], packagesWithScripts = new Map() }) {
+  if(missingPolicies.length) {
+    console.log('packages missing configuration:')
+    missingPolicies.forEach(pattern => {
+      const collection = packagesWithScripts.get(pattern) || []
+      console.log(`- ${pattern} [${collection.length} location(s)]`)
+    })
+  }
+}
+
+// internals
+
+/**
+ * 
+ * @param {Object} arg 
+ * @param {string} arg.event
+ * @param {Array<PkgInfo>} arg.packages
+ */
+async function runAllScriptsForEvent({ event, packages }) {
   for (const { canonicalName, path, scripts } of packages) {
     if (event in scripts) {
       console.log(`- ${canonicalName}`)
       await runScript({ path, event })
     }
   }
 }
+/**
+ * @param {Array<BinInfo>} allowedBins 
+ */
+async function installBinScripts(allowedBins) {
+  for (const { bin, path, link, canonicalName } of allowedBins) {
+    console.log(`- ${canonicalName}`)
+    await linkBinRelative({ path, bin, link, force: true })
+  }
```

Comment:
Any reason not to process these concurrently?

```suggestion
  return Promise.all(allowedBins.map(
    ({ bin, path, link, canonicalName }) => {
      console.log(`- ${canonicalName}`)
      return linkBinRelative({ path, bin, link, force: true })
    }
  )
```

Reply:
In case of an error, the names in potput would not help figure out which one failed. 
And they do run in parallel within linkBinRelative AFAIR when there's more than one bin in a package.

Reply:
Hm, I don't see how they run in parallel there - something additional in `bin-links`?

But either way I see the point on the message ordering - sounds reasonable to make that take priority :+1: